components/pedigree/resources/src/main/resources/pedigree/view/person.js

```diff
@@ -1579,6 +1579,8 @@ define([
                     //          {id: 'ENSG00000160688', gene: 'FLAD1', status: 'solved'},
                     //          {id: 'custom', 'gene': 'custom', status: 'candidate'} ]
                     this._setGenes(info.genes);
+                } else {
+                    this._setGenes([]);
```

Comment:
I can think of something like making an array map with properties and iterate over an array with generic function:

`var properties = {
        "adoptedStatus"   : {
           default   : "",
           get : this.getAdopted(),
           set : setAdopted()      
        }, ...`

and then call `for (var property in properties) {
                           if(info.hasOwnProperty(property)) {
                               if (this.get(properties[property]) != info[property]) {
                                   this.set(info[property]);
                               }
                           } else {
                               this.set(properties[property].default);
                           }`